components/rollup_node/p2p_interface.md

```diff
@@ -0,0 +1,206 @@
+# P2P Interface
+
+This peer-to-peer (P2P) network service in the rollup node is optional,
+but greatly improves the latency between the view of sequencers and the rest of the network,
+as well as enabling faster historical sync to be bootstrapped, without relying on a single endpoint.
+
+In summary, the P2P stack looks like:
+- Discovery to find peers: [Discv5][discv5]
+- Connections, peering, transport security, multi-plexing, gossip: [LibP2P][libp2p]
+
+This document only specifies the composition and configuration of these network libraries.
+These components have their own standards, implementations in Go/Rust/Java/Nim/JS/more,
+and adopted several other blockchains, most notably the [L1 consensus layer (Eth2)][eth2-p2p].
+
+## P2P configuration
+
+### Identification
+
+Nodes have a **separate** network- and consensus-identity.
+The network identity is a `secp256k1` key, used for both discovery and active LibP2P connections.
+
+Common representations of network identity:
+- `PeerID`: a LibP2P specific ID derived from the pubkey (through protobuf encoding, typing and hashing)
+- `NodeID`: a Discv5 specific ID derived from the pubkey (through hashing, used in the DHT)
+- `Multi-address`: an unsigned address, containing: IP, TCP port, PeerID
+- `ENR`: a signed record used for discovery, containing: IP, TCP port, UDP port, signature (pubkey can be derived) and L2 network identification. Generally encoded in base64.
+
+### Discv5
+
+#### Structure
+
+The Ethereum Node Record (ENR) for an Optimism rollup node must contain keyed values:
```

Comment:
What do you mean by "keyed" here?

Reply:
key-value pairs; each value is identified by a unique key

Reply:
I'll change it to key-value pairs if that reads better